<p> 1) Importe os dados e verifique as primeiras linhas do dataset. Verifique também as últimas linhas do dataset.</p>

In [44]:
import pandas as pd
df= pd.read_csv('Insurance.csv')
#verificando as primeiras linhas do dataset

df.head(5)

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [45]:
df.tail(10)

age     sex     bmi  children smoker     region      charges
1328   23  female  24.225         2     no  northeast  22395.74424
1329   52    male  38.600         2     no  southwest  10325.20600
1330   57  female  25.740         2     no  southeast  12629.16560
1331   23  female  33.400         0     no  southwest  10795.93733
1332   52  female  44.700         3     no  southwest  11411.68500
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

<p>2) Verifique se existe algum valor missing no dataset.</p>

In [46]:
#metodo 1 
#para verificar se existe algum valor missing no dataset
#não contem nulos

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [47]:
#metodo 2
#verificando valores nulos por coluna
valores_missing=df.isnull().sum()
print(valores_missing)

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64


<p>3) Quanto de memória o dataset está consumindo? </p>

<p> <b>resposta:</b> uma forma de verificar quanto de memória o dataset está consumindo também é utilizando a função df.info(). </p>

<p> Mas se você quiser uma visão mais detalhada use 'df.memory_usage() </p>

In [48]:
memoria_coluna= df.memory_usage(deep=True)
print(memoria_coluna)

Index         128
age         10704
sex         82942
bmi         10704
children    10704
smoker      79216
region      88308
charges     10704
dtype: int64


In [49]:
memoria_total=memoria_coluna.sum()
print(memoria_total)

293410


 <p> <b> exercício 4) </b> Gere as estatísticas descritivas das colunas numéricas do dataset. Qual a mediana da coluna BMI? E a média? Qual desses valores é maior? O que isso significa? Caso não saiba o que isso significa, passe os 2 valores ao chatGPT e pergunte a ele o que significa a média ser maior ou menor que a mediana </p>

In [50]:
df['bmi'].mean() #media da coluna



30.66339686098655

In [51]:
df['bmi'].median() #mediana

30.4

<p> A mediana indica que metade dos dados está dentro desse intervalo de valores, nesse caso de 30.4. E a média, indica que o número médio, é simplesmente 30.4. A média ser maior que a mediana implica que existe uma desigualdade, pois se a metade da distribuição contém até 30.4. E a média ainda ser 30.6, pode ser indícios de outliers que estão puxando a média para cima. Indicando que a distribuição está inclinada para direita. A média é bastante influenciada por grandes números </p>

<p> <b> 5) </b> Qual a proporção de fumantes no nosso dataset? E de não fumantes? </p>

In [52]:
df.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [53]:
#numero total
total_pessoas=df['smoker'].count()

#numero total de fumantes
total_fumantes=(df['smoker']=='yes').sum()

#numero total de não fumantes
total_nfumantes=(df['smoker']=='no').sum()

print(f'O total de fumantes é de: {total_fumantes}')
print(f'O total de não fumantes é de: {total_nfumantes}')
print(f' E o total de pessoas é de: {total_pessoas}')

O total de fumantes é de: 274
O total de não fumantes é de: 1064
 E o total de pessoas é de: 1338


<p> <b> 6) </b> Existe alguma diferença na cobrança para fumantes e não fumantes? Se sim, ao que você imagina que isso se deve? </p>

<p> É preciso ter em mente, que o df.head() não nos mostra todas as colunas do nosso dataframe, apenas as primeiras colunas. Para verificar se temos informações de preços, podemos utilizar df.columns or df.columns.tolist()

In [54]:
df.columns.tolist()

['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges']

<p> Charges refere a custos ou despesas, ou simplesmente taxas cobradas. Podemos utilizar essa variável para análisar. Podemos comparar pela média.

In [55]:
#media da taxa cobrada de fumantes
charges_smokers= df[df['smoker']=='yes']['charges'].mean()

#media para não fumantes

charges_non_smokers= df[df['smoker']=='no']['charges'].mean()

print(f'A média do charges para fumantes é de: {charges_smokers}')
print(f' A média do charges para não fumantes é de:{charges_non_smokers}')

A média do charges para fumantes é de: 32050.23183153284
 A média do charges para não fumantes é de:8434.268297856204


<p> Sim, fumantes possui um maior custo para o plano de saúde. Isso implica que utilizam mais o plano. O que faz sentido, pois o ato de fumar é prejudicial a saúde e está fortemente associada a diversas doenças. </p>

<p> <b> 7) </b> Qual a média de BMI para fumantes? Ela é muito diferente da observada para fumantes? Se sim, por qual motivo você acha que isso ocorre? </p>

<p> BMI é o nosso IMC, índice de massa corporal. </
p>

In [56]:
#BMI para fumantes
BMI_smokers= df[df['smoker']=='yes']['bmi'].mean()

#bmi para não fumantes
BMI_notsmokers= df[df['smoker']=='no']['bmi'].mean()

print(f'A média do bmi para fumantes foi de {BMI_smokers}')
print(f'A média de bmi para não fumantes foi de {BMI_notsmokers}')

A média do bmi para fumantes foi de 30.70844890510949
A média de bmi para não fumantes foi de 30.651795112781954


<p> E a saída foi de bmi para fumantes de 30.7 e 30.6 para não fumantes.
Resposta: O bmi médio em ambos os grupos foi muito próximo, não demonstrando uma diferença grande. Mas como ambas deram altas, a presença de obesos na amostra podem está aumentando o bmi médio. O mais adequado seria trabalhar com a média por faixas de bmi separando os dois grupos, ou realizar um test t para analisar a probalidade de distribuição. </p>

<p>8) Existe alguma diferença de cobrança para quem tem filhos? Identifique se ter filhos parece acarretar numa cobrança maior olhando a média de charges para quem tem 0 filhos e a média para quem tem 1 ou mais filhos.</p>

In [57]:
#obentendo as médias
mean_charges_nochildren=df[df['children']==0]['charges'].mean()
mean_charges_yeschildren=df[df['children']>=1]['charges'].mean()

#imprimindo as médias
print(f'A média de cobrança para quem possui 1 ou mais filhos é de {mean_charges_yeschildren: .2f}')
print(f'A média de cobrança para quem não tem filhos é de {mean_charges_nochildren: .2f}')

A média de cobrança para quem possui 1 ou mais filhos é de  13949.94
A média de cobrança para quem não tem filhos é de  12365.98


<p> Também podemos verificar qual a média maior de forma automática da seguinte forma </p>

In [58]:
#obtendo as médias
mean_charges_nochildren=df[df['children']==0]['charges'].mean()
mean_charges_yeschildren=df[df['children']>=1]['charges'].mean()

#verificando qual o maior e imprimindo
if mean_charges_nochildren > mean_charges_yeschildren:
    print("A média de cobrança para quem não tem filhos é maior.")
elif mean_charges_nochildren < mean_charges_yeschildren:
    print("A média de cobrança para quem tem 1 ou mais filhos é maior.")
else:
    print("As médias de cobrança para ambos os grupos são iguais.")

A média de cobrança para quem tem 1 ou mais filhos é maior.


<p> Foi observado que a média é maior para quem possui 1 ou mais filhos, isso é esperado, pois o plano de saúde costuma ter uma tarifa para cada dependente do plano de saúde. </p>

<p> 9) Para o homem, o BMI de 20 a 24,9 é considerado normal. Acima disso, é considerado acima do peso, abaixo é considerado abaixo do peso. Crie uma nova coluna, chamada "bmi_male" que receberá o texto "abaixo do normal" para BMI abaixo de 20, "normal" para BMI de 20 a 24,9 e "acima do normal" para BMI acima de 24,9. </p>

<p> Podemos resolver essa questão de diversas formas, mas a que acho mais interessante nesse caso é utilizar operações vetoriais </p>

<p> Operações vetoriais são operação que realizamos em toda coluna de uma vez, dispensando o uso de loops explícitos e sem precisar interar item por item. É bastante útil para atribuir valores em grandes conjutnos de dados </p>

<p> Mas por alguma razão não funcionou nesse exercício, então iremos matar a questão utilizando uma função lambda </p>